### PR TITLE
Removes @default_floating_dtype decorator

### PR DESCRIPTION
**Pull request by mruberry**
One fewer legacy decorator cluttering the test suite.

Functions relying on this decorator were updated or, in the case of test_sparse, the test suite was put back on double by default.

Note: this PR is blocked on #27599.

### CODE REVIEW

**Review by albanD — APPROVED** [review]

File: test/test_dataloader.py [hunk 1]
```diff
@@ -1484,8 +1484,8 @@ def check_len(dl, expected):
         check_len(DataLoader(self.dataset, batch_size=2), 50)
         check_len(DataLoader(self.dataset, batch_size=3), 34)
 
+    # FIXME: conversion from np.float64 to torch.DoubleTensor does not work
```
Comment (nairbv) [inline test/test_dataloader.py]: I think my changes in PR https://github.com/pytorch/pytorch/pull/30486 will fix it, is there an issue filed somewhere tracking this bug?

Comment (mruberry) [reply test/test_dataloader.py]: @nairbv cool! I couldn't find an issue for this, although I thought we had one filed. If I come across it I'll let you know.

**Pull request closed without merging**